src/detection_linux.cpp

```diff
@@ -89,16 +89,13 @@ void Stop() {
 		return;
 	}
 
-	isRunning = false;
-
 	uv_mutex_destroy(&notify_mutex);
 	uv_signal_stop(&int_signal);
 	uv_signal_stop(&term_signal);
 	uv_close((uv_handle_t *) &async_handler, NULL);
 	uv_cond_destroy(&notifyDeviceHandled);
-
-	udev_monitor_unref(mon);
-	udev_unref(udev);
+	
+	isRunning = false;
```

Comment:
Is there an exact behavior we're going for by moving this below (please explain and document)? Or is just stylistic preference?

I could see it maybe mattering if the `uv_xxx` stuff takes time to run and `Stop` is called multiple times, in the previous code, we wouldn't do all of the cleanup multiple times.

Reply:
yeah I agree that changing isRunning at the top makes more sense, as we want the loop to exit asap, and not start another iteration while we start cleaning up